File: clients/javascript/src/models/streamMsgOut.ts
// this file is @generated

export interface StreamMsgOut {
    offset: number;
    topic: string;
    value: number[];
    headers?: { [key: string]: string };
    timestamp: Date;
}

export const StreamMsgOutSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): StreamMsgOut {
        return {
            offset: object['offset'],
            topic: object['topic'],
            value: object['value'],
            headers: object['headers'],
            timestamp: new Date(object['timestamp']),
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: StreamMsgOut): any {
        return {
            'offset': self.offset,
            'topic': self.topic,
            'value': self.value,
            'headers': self.headers,
            'timestamp': self.timestamp,
        };
    }
}

File: clients/rust/src/models/stream_msg_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[non_exhaustive]
#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct StreamMsgOut {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub headers: Option<std::collections::HashMap<String, String>>,

    pub offset: u64,

    pub timestamp: jiff::Timestamp,

    pub topic: String,

    pub value: Vec<u8>,
}

impl StreamMsgOut {
    pub fn new(offset: u64, timestamp: jiff::Timestamp, topic: String, value: Vec<u8>) -> Self {
        Self {
            headers: None,
            offset,
            timestamp,
            topic,
            value,
        }
    }
}
